main/src/com/google/refine/grel/controls/ForRange.java

```diff
@@ -86,27 +83,31 @@ public Object call(Properties bindings, Evaluable[] args) {
                 long step = ((Number) stepO).longValue();
                 double to = ((Number) toO).doubleValue();
 
-                while (from < to) {
-                    bindings.put(indexName, from);
+                if (!((from < to && step > 0) || (from > to && step < 0))) {
```

Comment:
We want to make sure to check for `step == 0` here since that will cause an infinite loop, but I don't think we should be more permissive about parameters which will cause zero iterations e.g. from 0 to 10 by -1 because those values might be calculated and we don't want the callers to have to do special handling when zero iterations is an adequate solution.